pymeasure/instruments/mksinst/mks974b.py

```diff
@@ -0,0 +1,224 @@
+        validator=strict_discrete_set,
+        values=["ABOVE", "BELOW"],
+        check_set_errors=True,
+    )
+
+    enable = Channel.control(
+        "EN{ch}?", "EN{ch}!%s",
+        """Control the assigned input channel or disable the setpoint relay.""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={False: "OFF",
+                True: "ON",
+                "combined": "CMB",
+                "pirani": "PIR",
+                "piezo": "PZ",
+                "cold cathode": "CC",
+                },
+        check_set_errors=True,
+    )
+
+
+class MKS974B(MKSInstrument):
+    """ MKS 974B vacuum pressure transducer
+
+    Connection to the device is made through an RS232/RS485 serial connection.
+    The communication protocol of this device is as follows:
+
+    Query: '@<aaa><Command>?;FF' with the response '@<aaa>ACK<Response>;FF'
+    Set command: '@<aaa><Command>!<parameter>;FF' with the response '@<aaa>ACK<Response>;FF'
+    Above <aaa> is an address from 001 to 254 which can be specified upon
+    initialization. Since ';FF' is not supported by pyvisa as terminator this
+    class overloads the device communication methods.
+
+    :param adapter: pyvisa resource name of the instrument or adapter instance
+    :param string name: The name of the instrument.
+    :param address: device address included in every message to the instrument
+                    (default=253)
+    :param kwargs: Any valid key-word argument for Instrument
+    """
+
+    setpoint_1 = Instrument.ChannelCreator(SetpointChannel, 1)
+
+    setpoint_2 = Instrument.ChannelCreator(SetpointChannel, 2)
+
+    setpoint_3 = Instrument.ChannelCreator(SetpointChannel, 3)
+
+    def __init__(self, adapter, name="MKS 974B vacuum pressure transducer", address=253, **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            address=address,
+            **kwargs
+        )
+
+    def id(self):
+        """ Get the identification of the instrument. """
+        return f"{self.manufacturer}{self.model} {self.device_type} ({self.serial_number})"
+
+    serial_number = Instrument.measurement(
+        "SN?", """Get the serial number of the instrument""",
+        cast=str,
+    )
+
+    hardware_version = Instrument.measurement(
+        "HV?", """Get the hardware version of the instrument""",
+        cast=str,
+    )
+
+    firmware_version = Instrument.measurement(
+        "FV?", """Get the firmware version of the instrument""",
+        cast=str,
+    )
+
+    device_type = Instrument.measurement(
+        "DT?", """Get the device type""",
+        cast=str,
+    )
+
+    manufacturer = Instrument.measurement(
+        "MF?", """Get the manufacturer name""",
+        cast=str,
+    )
+
+    model = Instrument.measurement(
+        "MD?", """Get the transducer model number""",
+        cast=str,
+    )
+
+    part_number = Instrument.measurement(
+        "MD?", """Get the transducer part number""",
```

Comment:
Are these commands common to all instruments?
At least the serial number command is common.

If it is just the serial number, I like the current separation between communication logic in the parent and everything else in the child.

If it is a certain set of shared properties, they could be in the parent class.

Reply:
I now checked and for the MKS937B and 974B only the serial number is common. There is some minor overlap with respect to the RelayChannels/unit. I will look into it if there are options to still put few things to parent classes (and at the same time improve the 937B instrument -> does currently not have relay channels which it also supports)

Reply:
I kept the properties separate since I think the overlap is small. Even the unit property which one might assume to be the same is different in two minor details and I see no benefit in putting it to the central code.

Reply:
As you say: If the overlap is minimal, it is more work and more difficult to understand to put this stuff in a common superclass.